helm/prometheus-rules/templates/platform/atlas/alerting-rules/loki.rules.yml

```diff
@@ -83,3 +83,22 @@ spec:
         severity: page
         team: atlas
         topic: observability
+    - alert: LokiNeedsToBeScaledUp
+      annotations:
+        description: '{{`Loki component {{ $labels.horizontalpodautoscaler }} needs to be scaled up.`}}'
+        opsrecipe: loki/
+      expr: |
+        sum by (cluster_id, installation, namespace, pipeline, provider, horizontalpodautoscaler) (kube_horizontalpodautoscaler_status_desired_replicas{namespace="loki", horizontalpodautoscaler=~"loki-backend|loki-write|loki-read"})
+          != 
+        sum by (cluster_id, installation, namespace, pipeline, provider, horizontalpodautoscaler) (kube_horizontalpodautoscaler_status_current_replicas{namespace="loki", horizontalpodautoscaler=~"loki-backend|loki-write|loki-read"})
+      for: 10m
```

Comment:
As @hervenicol said I think 10 minutes is too short. I would probably make this 1h.

Also  I would think the alert should be named something like LokiHpaNotWorking or something like LokiHpaReplicasMismatch ?  I took it from here https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/3dfa72d1d1ab31a686b1f52ec28bbf77c972bd23/alerts/apps_alerts.libsonnet#L284

Reply:
I would also like if we were not paged if we reached the max number of replicas which is probably something we can get from the HPA?

Reply:
I am concerned by a specific scenario : someone has dramatically reduced resources requests for a loki component following a `LokiNeedsToBeScaledDown` alert but at some point, actual usage does increase which leads to the hpa scaling up the number of replicas. This would lead to having a lot of replicas with very low resources requests whereas we could have had less replcias with higher resources requests which is better right ?

Concerning the time period, I'm totally ok to increase it, 10m is indeed quite short. 

I'm also ok with renaming the alert.

Concerning your identical concern in the case of the component reaching the maxReplicas number I think it relates to my first paragraph right ? If the hpa still wants to scale up the component while being already at maxReplicas, this can mean we're in the scenario I described and that we should act, unless hpa can't recommend a replica number higer than it maxReplicas.

Reply:
> I am concerned by a specific scenario : someone has dramatically reduced resources requests for a loki component following a `LokiNeedsToBeScaledDown` alert but at some point, actual usage does increase which leads to the hpa scaling up the number of replicas. This would lead to having a lot of replicas with very low resources requests whereas we could have had less replcias with higher resources requests which is better right ?

In my opinion, this was exactly the reason why we wanted a "scale up" alert.
But I think your query does not match the use case.

> I'm also ok with renaming the alert.

Well, the proposed names are to match the current alert. But I'm not even sure we want the current alert. :sweat_smile: 

> If the hpa still wants to scale up the component while being already at maxReplicas, this can mean we're in the scenario I described and that we should act, unless hpa can't recommend a replica number higer than it maxReplicas.

Oh, so you want an alert that fires when the component reaches MaxReplicas but still requires more replicas?

I'm rather fine with this, except:

> on some clusters that are quite resource-tight, we may have voluntarily set a low maxReplicas as per the opsrecipe. So that alert would be constantly firing. What is the remediation for these cases? Set an infinite silence?

Reply:
Let's talk about all of this after a standup or during a dedicated meeting :sweat_smile:

Reply:
Following our discussions, I renamed the alert and increased the time period for triggering it to 4h.